crypto/s2n_pkey.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "crypto/s2n_ecdsa.h"
+#include "crypto/s2n_hash.h"
+#include "crypto/s2n_rsa.h"
+
+#include "utils/s2n_blob.h"
+
+/* Structure that models a public or private key and type-specific operations */
+struct s2n_pkey {
+    union {
+        struct s2n_rsa_key rsa_key;
+        struct s2n_ecdsa_key ecdsa_key;
+    } key;
+
+    const struct s2n_pkey_ctx *ctx;
+};
+
+struct s2n_pkey_ctx {
```

Comment:
nit: usually `*_ctx` structs are containers of data. Any reason why these function handlers need to be in a separate struct? We could just add them as fields in the `s2n_pkey` struct.